src/openshift/components/CryostatContainer.tsx

```diff
@@ -17,12 +17,55 @@ import * as React from 'react';
 import { store } from '@app/Shared/Redux/ReduxStore';
 import { Provider } from 'react-redux';
 import { CryostatController } from './CryostatController';
+import { CryostatContext, ServiceContext, Services } from '@app/Shared/Services/Services';
+import { TargetService } from '@app/Shared/Services/Target.service';
+import { SettingsService } from '@app/Shared/Services/Settings.service';
+import { LoginService } from '@app/Shared/Services/Login.service';
+import { ApiService } from '@app/Shared/Services/Api.service';
+import { NotificationsInstance } from '@app/Shared/Services/Notifications.service';
+import { NotificationChannel } from '@app/Shared/Services/NotificationChannel.service';
+import { ReportService } from '@app/Shared/Services/Report.service';
+import { TargetsService } from '@app/Shared/Services/Targets.service';
+import { pluginServices } from '@console-plugin/services/PluginContext';
+import { Observable } from 'rxjs';
+
+export const pluginContext: CryostatContext = {
+  url: (path?: string): Observable<string> => pluginServices.plugin.proxyUrl(path),
+  headers: () => {
+    const headers = new Headers({
+      // TODO populate these with context selections, maybe taken from redux or localstorage
+      'CRYOSTAT-SVC-NS': '',
+      'CRYOSTAT-SVC-NAME': '',
```

Comment:
Here we need to figure out something to link the Service selector context bar in. If we use redux or localstorage for this we can make sure that the selection is persisted even across the user closing and reopening the browser tab, however we would run into issues if the same user tries to use the plugin UI in two or more browser tabs at the same time. So, maybe the best thing to do is to simply hold this in client-side application memory. This context object could gain some public field to hold the selector pair, and this function would simply read the field to populate the returned headers.